src/domdiv/inline_images.py

```diff
@@ -0,0 +1,272 @@
+import re
+
+from . import resource_handling
+
+
+def expand_wiki_nowrap_template(s: str) -> str:
+    return re.sub(r"{{nowrap\|([^{}]+)}}", r"<nobr>\1</nobr>", s)
+
+
+def nobr(s: str) -> str:
+    return f"<nobr>{s}</nobr>"
+
+
+def nobr_if_necessary(s: str, needed: bool) -> str:
+    if needed:
+        return nobr(s)
+    else:
+        return s
+
+
+class InlineImage(object):
+    def __init__(self, width, height, valign):
+        self.width = width
+        self.height = height
+        self.valign = valign
+
+
+class CoinInlineImage(InlineImage):
+    def __init__(self, width, height, valign, resolution):
+        super().__init__(width, height, valign)
+        if resolution == "low":
+            # TODO: unlike the other low-res images, the empty coin doesn't have extra whitespace on the right,
+            # so it looks squashed when rendered at the aspect ratio that works for the others. It would be good to
+            # clean them all up at some point.
+            self.blank_image = "coin_small_empty.png"
+            self.question_image = "coin_small_question.png"
+            self.x_image = "coin_small_x.png"
+            self.n_image = "coin_small_{n}.png"
+        elif resolution == "medium":
+            self.blank_image = "coin.png"
+            self.question_image = "coin_small_question.png"
+            self.x_image = "coin_small_x.png"
+            self.n_image = "coin_small_{n}.png"
+        elif resolution == "high":
+            self.blank_image = "highres/Coin.png"
+            self.question_image = "highres/CoinQ.png"
+            self.x_image = "highres/Coinx.png"
+            self.n_image = "highres/Coin{n}.png"
```

Comment:
These are higher resolution images I downloaded from the wiki. I'm not quite sure where to land on the tradeoff between resolution and generated file size. This should get clarified before merging, either not referring to files that aren't committed, or committing them.